balancer-js/src/test/lib/utils.ts

```diff
@@ -287,6 +298,45 @@ export class TestPoolHelper {
   }
 }
 
+/**
+ * Returns a pool from the json file as a Pool type defined in SubgraphPoolRepository.
+ *
+ * @param id pool ID
+ * @param network we only support 1 and 137
+ * @returns Pool as from the SubgraphPoolRepository
+ */
+export const getPoolFromFile = async (
+  id: string,
+  network: 1 | 137
+): Promise<Pool> => {
+  const pool = await new PoolsJsonRepository(jsonPools[network], network).find(
+    id
+  );
+  if (pool === undefined) throw new Error('Pool Not Found');
+  return pool;
+};
+
+/**
+ * Updates pool balances with onchain state.
+ *
+ * @param pool pool from repository
+ * @param network we only support 1, 137 and 42161
+ * @returns Pool as from the SubgraphPoolRepository
+ */
+export const updateFromChain = async (
+  pool: Pool,
+  network: 1 | 137 | 42161,
+  provider: JsonRpcProvider
+): Promise<Pool> => {
+  const onChainPool = await getOnChainBalances(
+    [pool],
+    BALANCER_NETWORK_CONFIG[network].addresses.contracts.multicall,
+    BALANCER_NETWORK_CONFIG[network].addresses.contracts.vault,
+    provider
+  );
+  return onChainPool[0];
```

Comment:
How will this behave if we're providing a pool that exists on the Subgraph file, but not on chain?
A scenario where this could happen is when testing against an old blockNumber where the pool wasn't yet created.
Will it throw an error or will it simply not update the pool state?

Reply:
is there a pool like that? yes, it will throw. Onchain sync is mostly helpful in cases where some state changes need to be tested. When the pool is not onchain, then the whole spec wouldn't work.